src/transport/mqtt/config/mod.rs

```diff
@@ -402,14 +402,25 @@ where
             .map_err(|err| MqttError::Pairing(PairingError::InvalidUrl(err)))?;
 
         let insecure_ssl = self.ignore_ssl_errors || is_env_ignore_ssl();
-        let provider = TransportProvider::new(
+        let mut provider = TransportProvider::new(
             pairing_url,
             secret.clone(),
             builder.writable_dir.clone(),
             insecure_ssl,
         );
```

Comment:
We can add a method `configure`, called here, that will read the cert store and return the `TransportProvider`.

Reply:
This way we can remove the `Option` around the Cert Store and the error returned by the api_tls_config call.